src/ast/semantic_analyser.cpp

```diff
@@ -277,15 +277,13 @@ void SemanticAnalyser::visit(Builtin &builtin)
       auto matches = bpftrace_.find_wildcard_matches(attach_point->target,
                                                      attach_point->func,
                                                      *symbol_stream);
-      for (auto &match : matches) {
-        std::string tracepoint_struct = TracepointFormatParser::get_struct_name(
-            attach_point->target, match);
-        Struct &cstruct = bpftrace_.structs_[tracepoint_struct];
-        builtin.type = SizedType(Type::cast, cstruct.size, tracepoint_struct);
-        builtin.type.is_pointer = true;
-        builtin.type.is_tparg = true;
-        break;
-      }
+      auto &match = *matches.begin();
```

Comment:
I think we need to check that matches were found before we deref this iterator.

Reply:
https://github.com/iovisor/bpftrace/blob/45cbee56cc007be2030c1f9d368764d0e8a95fad/src/ast/semantic_analyser.cpp#L281-L291